### PR TITLE
Modified version of BaseProvider.cs

### PR DESCRIPTION
**Pull request by SuperCreedy**
Changes to BaseProvider.cs to resolve an issue with multiple SocketExceptions being logged on a production site.

Error message was: System.Net.Sockets.SocketException: Only one usage of each socket address (protocol/network address/port) is normally permitted.

We've been running with this modified code for a couple of months now and our servers have been much more stable.

Hope this helps!

References:
https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/
http://blog.trivident.com/tuning-dd4t-rest/

### CODE REVIEW

**Comment by quirijnslings** [comment]
Change looks good and makes total sense.
I do have a question about the lifetime of the http client and related classes in the provider. These objects are created in the constructor, which means the lifetime depends on the DI settings. From what I see, it looks like Autofac will instantiate it with its default lifetime (Per Dependency, which in this case comes down to 'forever', I think - see http://autofaccn.readthedocs.io/en/latest/lifetime/instance-scope.html).

This is where the providers are instantiated: https://github.com/dd4t/DD4T.DI.Autofac/blob/develop/source/DD4T.DI.Autofac/Providers.cs

And this method is normally called from the Application_Start method in the Global.asax, in other words: only once.

So to summarize: I think this HttpClient, HttpMessageHandler and HttpClientHandler are all created at application start, and used forever. @kunalshetye do you think there is a problem with that at all? It's fine, right? Perhaps @sshibani would like to comment as well?

**Comment by kpmrafeeq** [comment]
@quirijnslings , I had a couple of issues making HttpClient as a singleton, please refer http://byterot.blogspot.ae/2016/07/singleton-httpclient-dns.html
There are issues if we are using multiple instances as well https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/

**Comment by quirijnslings** [comment]
As I don't know how often our customers are really affected by this DNS change issue, and since this is a very real improvement, I will merge the PR now and look at the ConnectionLeaseTimeout later.
